lib/adapters/fetch.js

```diff
@@ -17,11 +17,11 @@ const fetchProgressDecorator = (total, fn) => {
   }));
 }
 
-const isFetchSupported = typeof fetch !== 'undefined';
-const isReadableStreamSupported = isFetchSupported && typeof ReadableStream !== 'undefined';
+const isFetchSupported = typeof fetch === 'function' && typeof Request === 'function' && typeof Response === 'function';
+const isReadableStreamSupported = isFetchSupported && typeof ReadableStream === 'function';
```

Comment:
Sorry for my 5 cent, but having `isFunction` utility will be more concise, more readable, and guard against typos (`typeof fetch === 'funct1on'` is just silently `false`), as well as increase evolvability, when additional sub-checks should be implemented.